scripts/switch.py

```diff
@@ -304,78 +314,62 @@ def __init__(self,
-        self.cutout = cutout
+        self.mx_cutout = cutout
 
-        _name = name + '_' + switch_type
-        _description = description + ' ' + switch_type + ' Mount'
-        _tags = tags + ' ' + switch_type
+        _name = f'SW_Cherry_MX_{switch_type}'
 
-        Switch.__init__(self,
+        super().__init__(
                         name=_name,
-                        description=_description,
-                        tags=_tags,
+                        description=f'Cherry MX keyswitch {switch_type} Mount',
+                        tags=f'Cherry MX Keyboard Keyswitch Switch {switch_type}',
                         cutout=True if cutout is not None else False,
-                        keycap=keycap,
-                        path3d=path3d,
-                        model3d=model3d if model3d is not None
-                        else f'{_name}.wrl')
+                        model3d=f'{_name}.wrl',
+                        **kwargs)
 
         self._init_switch()
 
-        if cutout is not None:
-            if cutout == 'simple':
-                self._init_cutout_simple()
-            elif cutout == 'relief':
-                self._init_cutout_relief()
-
-        if keycap is not None:
-            self.append(keycap)
-
-    def _init_switch(self):
-        self._init_fab_outline()
-        self._init_silkscreen()
-        self._init_courtyard()
-        self._init_tht_pads()
-        self._init_center_hole()
-
+    def _init_pcb_mount_holes(self):
         if self.switch_type == 'PCB':
-            self._init_pcb_mount_holes()
+            super()._init_pcb_mount_holes()
 
 
 # https://www.cherrymx.de/en/dev.html
 # https://github.com/keyboardio/keyswitch_documentation/blob/master/datasheets/ALPS/SKCL.pdf
-class SwitchHybridCherryMxAlps(Switch):
+class SwitchHybridCherryMxAlps(Switch, CherryMX):
 
     cherry_w = 14
     cherry_h = 14
 
     alps_w = 15.5
     alps_h = 12.8
 
+    base_polyline = [[-cherry_w/2, -cherry_h/2],
+                     [cherry_w/2, -cherry_h/2],
+                     [cherry_w/2, -alps_h/2],
+                     [alps_w/2, -alps_h/2],
+                     [alps_w/2, alps_h/2],
+                     [cherry_w/2, alps_h/2],
+                     [cherry_w/2, cherry_h/2],
+                     [-cherry_w/2, cherry_h/2],
+                     [-cherry_w/2, alps_h/2],
+                     [-alps_w/2, alps_h/2],
+                     [-alps_w/2, -alps_h/2],
+                     [-cherry_w/2, -alps_h/2],
+                     [-cherry_w/2, -cherry_h/2]]
+
     def __init__(self,
-                 name: str = 'SW_Hybrid_Cherry_MX_Alps',
-                 description: str = 'Cherry MX / Alps keyswitch hybrid',
-                 tags: str = 'Cherry MX Alps Matias Hybrid Keyboard Keyswitch'
-                             'Switch PCB',
-                 keycap: Keycap = None,
-                 path3d: str = None, model3d: str = None):
+                 path3d: str = None, model3d: str = None,
```

Comment:
This didn't quite work because of how the parent class handles path3d (namely, it isn't stored). Last commit takes care of this by allowing a multivalued model3d.